game-core/src/main/java/games/strategy/engine/data/ModifiableProperty.java

```diff
@@ -0,0 +1,145 @@
+package games.strategy.engine.data;
+
+import java.util.function.Supplier;
+
+/**
+ * A wrapper interface to Bundle setters, getters and resetters of the same field.
+ *
+ * @param <T> The type of the field to set, get and reset.
+ */
+public interface ModifiableProperty<T> {
```

Comment:
Not sure this is an improvement, but `Mutable` may be a better adjective than `Modifiable`.